src/redprl/refiner.fun

```diff
@@ -637,7 +637,7 @@ struct
          | (Syn.PROJ _, Syn.PROJ _) => Lcf.rule o Record.EqProj (* XXX should consult autoSynthesizableNeu *)
          | (Syn.DIM_APP (_, r1), Syn.DIM_APP (_, r2)) =>
            (case (Abt.out r1, Abt.out r2) of 
-               (`_, `_) => Lcf.rule o Path.EqApp orelse_ Lcf.rule o Line.EqApp
```

Comment:
The `orelse_` is useless, since the first one will simply succeed; if it was a line, we don't know it yet, and we will be stuck. Naturally we would want to use backtracking here, but this blows up the search space sadly, and our examples no longer terminate in reasonable time.